File: day_1/day_1.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

vector<int> readFile() {
    vector<int> data;

    ifstream in("day_1_input.txt", ios::in);

    int temp;

    while(in >> temp){
        data.push_back(temp);
    }

    in.close();
    
    return data;
}

int stageOne() {
    vector<int> data = readFile();
    int counter = 0;

    for (int x = 0; x < data.size() - 1; x++)
    {
        if (data[x] < data[x+1])
        {
            counter++;
        }
    }
    return counter;
}

int main(){
    cout << " - Stage One: " << stageOne() << endl;

    return 0;
}

File: day_1/day_1.py
def readFile():
    f = open("day_1_input.txt", "r")
    data = [int(line.strip()) for line in f]
    f.close()
    return data

def puzzleOne():
    counter = 0
    data = readFile()
    for x in range(1, len(data)):
        if data[x] > data[x-1]:
            counter += 1
    
    print("Part 1: " + counter)


if __name__ == '__main__':
    puzzleOne()
